### PR TITLE
switch networks POC

### PR DESCRIPTION
**Pull request by patnir**
Ability to switch networks from mainnet to testnet
- New component to add a banner when on testnet
- Styling changes to the network selector and language selector
- Redirects user to home when the network is switched to download snapshot or create an account

Figma: https://www.figma.com/file/fYw3aNgfS0qCS2jezGqv8P/Node-App-(New-Ui-Kit)?type=design&node-id=7785-3417&mode=design&t=AuNLp2QtvepagYNl-0

### CODE REVIEW

**Comment by patnir** [comment]
@dgca Lmk what you think, the switching state isn't the "smoothest", I tried a few options and this was the best. Maybe the whole app should go to a "Switching networks" loading page? Trying to decide if what we have is good enough to deploy

File: renderer/components/TestnetBanner/TestnetBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    defaultMessage: "You are currently on Testnet!",
+  },
+});
+
+export function TestnetBanner() {
+  const { data } = trpcReact.getStatus.useQuery();
+  const { formatMessage } = useIntl();
+
+  if (data?.node.networkId === 1) {
+    return <></>;
```
Comment (dgca) [inline renderer/components/TestnetBanner/TestnetBanner.tsx:18]: nit: `return null` is the more common way to do this.

File: renderer/components/NetworkSelector/NetworkSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  const { formatMessage } = useIntl();
+  const [selectedNetwork, setSelectedNetwork] = useState(initialNetwork);
+  const [isNetworkChanged, setIsNetworkChanged] = useState(false);
+
+  useEffect(() => {
+    if (isSuccess) {
+      router.push("/home");
+      onClose();
+    }
+  }, [onClose, isSuccess, router]);
```
Comment (dgca) [inline renderer/components/NetworkSelector/NetworkSelector.tsx:167]: suggestion: the mutation function (`changeNetwork`) can take a second argument with `{ onSuccess, onError, ... }` that you could use to do this instead. Probably a bit cleaner than using `useEffect`.

The onClick would look something like:
```
onClick={() => {
  changeNetwork({
    network: String(selectedNetwork) as "MAINNET" | "TESTNET",
  }, {
    onSuccess: () => {
      router.push('/home');
      onclose();
    }
  });
}}
```

**Comment by dgca** [comment]
IIRC, I thought we wanted to persist the selected network, it looks like it's always opening to mainnet right now.

File: renderer/components/NetworkSelector/NetworkSelector.tsx (hunk 2 above)
Comment (patnir) [reply renderer/components/NetworkSelector/NetworkSelector.tsx:167]: Ah nice, I'll keep this pattern in mind

File: main/api/snapshot/index.ts [hunk 3]
```diff
@@ -36,6 +36,10 @@ export const snapshotRouter = t.router({
     }),
   downloadSnapshot: t.procedure.mutation(async () => {
     const ironfish = await manager.getIronfish();
+    if (ironfish.isStarted()) {
+      await ironfish.stop();
+      await ironfish.init();
+    }
```
[the remark this answers is not in the document]
Comment (patnir) [reply main/api/snapshot/index.ts:42]: I'll look into a better way to do this, maybe something in the manager

**Comment by patnir** [comment]
Converting this to draft to "persist the selected network"

File: renderer/pages/home.tsx [hunk 4]
```diff
@@ -11,11 +11,21 @@ import { LogoLg } from "@/ui/SVGs/LogoLg";
  */
 export default function Home() {
   const router = useRouter();
+
+  const [randomValue] = useState(() => {
+    return Math.random();
+  });
```
Comment (dgca) [inline renderer/pages/home.tsx:17]: suggestion: Maybe add a comment here to explain the caching issue.

**Review by dgca — APPROVED** [review]

**Pull request merged**